README.md

```diff
@@ -102,6 +103,90 @@ rules:
   verbs: ["get", "list", "watch", "create", "update", "patch", "delete"]
 ```
 
+### AggregateReconciler
+
+An [`AggregateReconciler`](https://pkg.go.dev/github.com/vmware-labs/reconciler-runtime/reconcilers#AggregateReconciler) is responsible for synthesizing a single resource, aggregated from other state. The AggregateReconciler is a fusion of the [ResourceReconciler](#resourcereconciler) and [ChildReconciler](#childreconciler). Instead of operating on all resources of a type, it will only operate on a specific resource identified by the type and request (namespace and name). Unlike the child reconciler, the "parent" and "child" resources are the same.
+
+The resource reconciler is responsible for:
+- fetching the resource being reconciled
+- creating a stash to pass state between sub reconcilers
+- passing the resource to each sub reconciler in turn
+- creates the resource if it does not exist
+- updates the resource if it drifts from the desired state
+- deletes the resource if no longer desired
+- logging the reconcilers activities
+- records events for mutations and errors
+
+The implementor is responsible for:
+- specifying the type, namespace and name of the aggregate resource
+- defining the desired state
+- indicating if two resources are semantically equal
+- merging the actual resource with the desired state (often as simple as copying the spec and labels)
+
+**Example:**
+
+Resource reconcilers tend to be quite simple, as they delegate their work to sub reconcilers. We'll use an example from projectriff of the Function resource, which uses Kpack to build images from a git repo. In this case the FunctionTargetImageReconciler resolves the target image for the function, and FunctionChildImageReconciler creates a child Kpack Image resource based on the resolve value. 
+
+```go
```

Comment:
**[thought/suggestion]** I admit that I find the example too abstract. I have a hard time constructing a mental representation of what's going on and finding myself imagining what `Derive...` and `Resolve...` do. That's not what I would want from an example.

Maybe the example could be "dumbed down".

I can't quite eatablish the relationship between `DesiredResource` and `Reconcilers`, what get's called when and if they interact.

Reply:
I replaced this with a paragraph actually about the example.